src/refdb_fs.c

```diff
@@ -1246,9 +1246,17 @@ static int maybe_append_head(
 	if (ref->type == GIT_REFERENCE_SYMBOLIC)
 		return 0;
 
-	if ((error = git_reference_lookup(&head, backend->repo, GIT_HEAD_FILE)) < 0)
+	if ((error = git_reference_lookup(&head, backend->repo, GIT_HEAD_FILE))
+	    && error != GIT_ENOTFOUND)
 		return error;
 
+	/* Our repo is somehow missing its HEAD. Ignore the reflog in that case */
+	if (error == GIT_ENOTFOUND) {
+		/* XXX: warn ? */
+		git_error_clear();
+		return 0;
+	}
```

Comment:
```
if ((error = git_reference_lookup(&head, backend->repo, GIT_HEAD_FILE)) < 0) {
    /* Our repo is somehow missing its HEAD. Ignore the reflog in that case */
    if (error == GIT_ENOTFOUND)
         error = 0;
    return error;
}
````

Reply:
I think we also don't need to `git_error_clear()`. We state that errors are only valid immediately after a call to libgit2 that's returned an error.

Reply:
This is a workaround for the test case that explicitely removes `HEAD` and expect the refdb to keep working. This makes a missing `HEAD` be silently ignored (and the warn comment is there to remind me that I need to warn about that when warning support will have landed).

I'll use your version though.